src/bin/pg_autoctl/cli_do_misc.c

```diff
@@ -227,13 +227,19 @@ keeper_cli_create_replication_user(int argc, char **argv)
 void
 keeper_cli_pgsetup_discover(int argc, char **argv)
 {
+	bool missingPgdataOk = true;
 	PostgresSetup pgSetup = { 0 };
 
 	if (!pg_setup_init(&pgSetup, &keeperOptions.pgSetup, true, true))
 	{
 		exit(EXIT_CODE_PGCTL);
 	}
 
+	if (!pg_controldata(&pgSetup, missingPgdataOk))
+	{
+		exit(EXIT_CODE_PGCTL);
+	}
```

Comment:
Why is this needed now?

Reply:
Because we don't call `pg_controldata` from `pg_setup_init` as we used to in another PR. Baring that the output of the `pg_autoctl do pgsetup discover` commands are always having zeroes for the pg_control fields.